pkg/datapath/iptables/iptables.go

```diff
@@ -1564,7 +1564,11 @@ func (m *IptablesManager) installRules(ifName string) error {
 	}
 
 	if skipPodTrafficConntrack(false) {
-		podsCIDR := option.Config.GetIPv4NativeRoutingCIDR().String()
+		nativeRoutingCIDR := option.Config.GetIPv4NativeRoutingCIDR()
+		if nativeRoutingCIDR == nil {
+			return fmt.Errorf("the IPv4NativeRoutingCIDR is nil")
+		}
```

Comment:
Let's discuss in the thread.

Could we check the nil pointer inside CIDR.String()? For example:

```go
func (cidr *CIDR) String() string {
    if cidr == nil {
        return "<nil>"
    }
    return cidr.IPNet.String()
}
```
So we won't be bothered by nil pointer of *CIDR at other places.

Reply:
I got it